content/building-blocks/automate-and-execute-your-work/reproducible-work/dockerhub.md

```diff
@@ -0,0 +1,200 @@
+
+In case you are new to Docker it is recommended that you review some of the basics of Docker before going ahead with this building block. You can do that by visiting TSH's building block on [Docker for reproducible research](https://tilburgsciencehub.com/building-blocks/automate-and-execute-your-work/reproducible-work/docker/), there you will also learn more on how to create your Dockerfiles and build images from them and more!
+
+{{% /tip %}}
+
+### Your environment's dockerfile
+
+The starting point for any docker image is a dockerfile, which in this case will contain the instructions to set up in place all the dependencies associated with your environment.
+
+{{% codeblock %}}
+
+Example dockerfile
+
+```dockerfile
+FROM continuumio/miniconda3
+
+RUN conda install jupyter
+RUN pip install pytest
+RUN mkdir /your-env
+```
+
+{{% /codeblock %}}
+
+Above you can see an example of a simple dockerfile suited for this building block. As with any dockerfile, it starts with the `FROM` instruction, which indicates which [parent image](https://docs.docker.com/build/building/base-images/) is going to be used. In this case, the parent image is a popular one that comes with [Miniconda](https://docs.conda.io/en/latest/miniconda.html) pre-installed, a minimalist version of the python distribution platform for data science [Anaconda](https://www.anaconda.com/). One of the main advantages of employing this parent image is that it will already contain the elemental components to run python on it without including too many additional elements which may not be of interest for a particular project and may make the container unnecessarily large.  
+
+After that the `RUN` instruction, which is used to execute commands within the container, is employed to install jupyter notebook as well as any packages that are necessary to replicate your environment. In this particular example, the "pytest" package is the only one that is going to be downloaded. Note that miniconda includes already some of the most widely used python packages such as Numpy and Pandas
+
+Lastly, the `RUN` instruction is employed again to create a new directory within the image's container that will be used to connect the container filesystem with your pc filesystem. Don't worry if you don't fully grasp this now, you will come back at this later in the building block. 
+
+Feel free to use this example dockerfile as a template to export your environment if it adjusts to your needs! For that, you should add to it the name of the packages included in your environment (instead of the included pytest) and save it as a text file without an extension named "dockerfile" (i.e. the file should be named "dockerfile" not "dockerfile.txt" or anything else, otherwise docker will not recognize it.) 
```

Comment:
Maybe we can also say that they don't have to manually write the names of the all the packages here - if they have an `environment.yml` file with all the packages listed (as is common with conda usage), they can also do something like `conda create <filename>`

Reply:
Cool! I've included a mention to this option in the BB as a tip.